src/nvme/mi.c

```diff
@@ -419,6 +418,80 @@ int nvme_mi_admin_xfer(nvme_mi_ctrl_t ctrl,
 	return 0;
 }
 
+int nvme_mi_admin_admin_passthru(nvme_mi_ctrl_t ctrl, __u8 opcode, __u8 flags,
+				 __u16 rsvd, __u32 nsid, __u32 cdw2, __u32 cdw3,
+				 __u32 cdw10, __u32 cdw11, __u32 cdw12,
+				 __u32 cdw13, __u32 cdw14, __u32 cdw15,
+				 __u32 data_len, void *data, __u32 metadata_len,
+				 void *metadata, __u32 timeout_ms, __u32 *result)
+{
+	/* Input parameters flags, rsvd, metadata, metadata_len are not used */
+	struct nvme_mi_admin_resp_hdr resp_hdr;
+	struct nvme_mi_admin_req_hdr req_hdr;
+	struct nvme_mi_resp resp;
+	struct nvme_mi_req req;
+	int rc;
+	if(data_len > 4096) {
+		nvme_msg(ctrl->ep->root, LOG_ERR, "nvme_mi_admin_admin_passthru"
+			"does not supports data_len over 4096 bytes.\n");
+		errno = EINVAL;
+		return -1;
+	}
+
+	if(timeout_ms > nvme_mi_ep_get_timeout(ctrl->ep)) {
```

Comment:
space after the `if` here; can you run this through checkpatch?

Reply:
Fixed and attached checkpatch.pl result in the comment above.